cmd/zt_traverser_blob_test.go

```diff
@@ -61,6 +65,989 @@ func TestIsSourceDirWithStub(t *testing.T) {
+func TestIsDirWithList(t *testing.T) {
+	a := assert.New(t)
+	//create mock server
+	srv, close := mock_server.NewServer(mock_server.WithTransformAllRequestsToTestServerUrl())
+	defer close()
+
+	//initialize variables
+	accountName, rawURL, blobURL, _, cName, credential, err := SetUpVariables()
+	a.Nil(err)
+
+	client, err := blobservice.NewClientWithSharedKeyCredential(rawURL, credential,
+		&blobservice.ClientOptions{
+			ClientOptions: azcore.ClientOptions{
+				Transport: srv, //passing in mock server
+			}})
+	a.Nil(err)
+
+	//get properties returns error
+	srv.AppendResponse(mock_server.WithStatusCode(404), mock_server.WithHeader("x-ms-error-code", "BlobNotFound"), mock_server.WithBody([]byte(MockErrorBody("BlobNotFound"))))
+	//list returns blob items
+	srv.AppendResponse(mock_server.WithStatusCode(200), mock_server.WithBody([]byte(SetBodyWithBlob(accountName, cName))))
+
+	//create new blob traverser
+	blobTraverser := newBlobTraverser(blobURL, client, ctx, true, true, func(common.EntityType) {}, false, common.CpkOptions{}, false, false, false, common.EPreservePermissionsOption.None(), false)
+
+	//test method and validate
+	isDir, err := blobTraverser.IsDirectory(true)
+	a.True(isDir)
+	a.Nil(err)
+}
+
+// this test calls Traverse and getPropertiesIfSingleBlob and responds with the BlobUsesCustomerSpecifiedEncryption error
+func TestTraverseGetPropErrorCode(t *testing.T) {
+	a := assert.New(t)
+
+	//create mock server
+	srv, close := mock_server.NewServer(mock_server.WithTransformAllRequestsToTestServerUrl())
+	defer close()
+
+	//initialize variables
+	_, rawURL, blobURL, _, _, credential, err := SetUpVariables()
+	a.Nil(err)
+
+	client, err := blobservice.NewClientWithSharedKeyCredential(rawURL, credential,
+		&blobservice.ClientOptions{
+			ClientOptions: azcore.ClientOptions{
+				Transport: srv, //passing in mock server
+			}})
+	a.Nil(err)
+
+	//the mock server returns "BlobUsesCustomerSpecifiedEncryption"
+	srv.AppendResponse(mock_server.WithStatusCode(409), mock_server.WithHeader("x-ms-error-code", "BlobUsesCustomerSpecifiedEncryption"), mock_server.WithBody([]byte(MockErrorBody("BlobUsesCustomerSpecifiedEncryption"))))
+
+	//create new blob traverser
+	blobTraverser := newBlobTraverser(blobURL, client, ctx, true, true, func(common.EntityType) {}, false, common.CpkOptions{}, false, false, false, common.EPreservePermissionsOption.None(), false)
+
+	//test method and validate
+	seenFiles := make(map[string]StoredObject)
+
+	err = blobTraverser.Traverse(nil, func(storedObject StoredObject) error {
+		seenFiles[storedObject.relativePath] = storedObject
+		return nil
+	}, []ObjectFilter{})
+	a.Equal(map[string]StoredObject{}, seenFiles)
+	a.Equal("this blob uses customer provided encryption keys (CPK). At the moment, AzCopy does not support CPK-encrypted blobs. If you wish to make use of this blob, we recommend using one of the Azure Storage SDKs", err.Error())
```

Comment:
This might be out of scope, but for these more specific errors, perhaps we should just have a `zt_traverser_errors.go` file where we store all of these strings in one place, that way, if it gets used in multiple places, or changed in the future, it's a zero-brainpower operation to find all the places to change it in tests and such.